refactor: rename `connection` to `grpc_channel`

ramontayag opened the pull request:
Since what is returned is a `GRPC.Channel`, use `grpc_channel` to more closely align with what is being worked with.

ramontayag committed:
ci: update workflow

Use new maintained https://github.com/erlef/setup-beam

RooSoft reviewed (commented):
I'd like to know why connections get renamed to channels...

It may make sense, but there already is a concept of channel in Lightning network. It seems to me that we now have two things that might end up having the same name.

Can you elaborate on this, please?

Thanks!

ramontayag commented:
I see. You're right, I didn't consider Lightning channels.

Perhaps `grpc_channel` is best because it leaves no doubt.

ramontayag committed:
refactor: rename `connection` to `grpc_channel`

Since what is returned is a `GRPC.Channel`, use
`grpc_channel` to more closely align with what is being worked with.

There's [also a concept of `channel` in Lightning](https://github.com/RooSoft/lnd_client/pull/17#pullrequestreview-1220836136),
therefore don't use plain `channel`

RooSoft commented:
Might be better, but still, I'd like to know why that change...
